qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/ConvTemplateBrowser.cpp

```diff
@@ -0,0 +1,318 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "ConvTemplateBrowser.h"
+
+#include "MantidAPI/FunctionFactory.h"
+#include "MantidAPI/IFunction.h"
+
+#include "MantidQtWidgets/Common/QtPropertyBrowser/ButtonEditorFactory.h"
+#include "MantidQtWidgets/Common/QtPropertyBrowser/CompositeEditorFactory.h"
+#include "MantidQtWidgets/Common/QtPropertyBrowser/DoubleEditorFactory.h"
+#include "MantidQtWidgets/Common/QtPropertyBrowser/qteditorfactory.h"
+#include "MantidQtWidgets/Common/QtPropertyBrowser/qtpropertymanager.h"
+#include "MantidQtWidgets/Common/QtPropertyBrowser/qttreepropertybrowser.h"
+
+#include <QMessageBox>
+#include <QSettings>
+#include <QVBoxLayout>
+
+#include <iostream>
+#include <limits>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace IDA {
+
+namespace {
+class ScopedFalse {
```

Comment:
Needs some documentation explaining it to suppress signals whilst the class is in scope